bundle/edu.gemini.spModel.io/src/main/scala/edu/gemini/spModel/io/SpImportService.scala

```diff
@@ -44,7 +44,8 @@ object SpImportService {
         }
       }
 
-      val cp = db.getFactory.copyWithNewKeys(im, POTUtil.getUnusedProgramId(im.getProgramID, db))
+      val pid = Option(im.getProgramID).map(POTUtil.getUnusedProgramId(_, db)).orNull
```

Comment:
I'm a bit confused about the `orNull` here. I guess we pass `null` into `db.getFactory.copyWithNewKeys` if we have no ID?

Reply:
Correct, and so it just doesn't assign an ID in that case.  It's suboptimal but this is code that dates back to the mid-1990s.